src/PuJetIdSF.cc

```diff
@@ -134,16 +142,69 @@ float PuJetIdSF::getSF(bool isReal, float pt, float eta)
   return sf;
 }
 
+// getSFError
+float PuJetIdSF::getSFError(bool isReal, float pt, float eta)
+{
+  float sferr;
+
+  if (pt<20.) pt=20.;
+  if (pt>50.) pt=50.;
+
+  if (isReal)
+  {
+    int binglobal = h_eff_SF_err_->FindBin (pt, eta);
+    int binx, biny, binz;
+    h_eff_SF_err_->GetBinXYZ (binglobal, binx, biny, binz); // converts to x, y bins
+    int nx = h_eff_SF_err_->GetNbinsX();
+    int ny = h_eff_SF_err_->GetNbinsY();
 
-//getEvtWeight
-float PuJetIdSF::getEvtWeight(bigTree &theBigTree, TLorentzVector tau1, TLorentzVector tau2)
+    // under-overflows
+    if (binx < 1) binx = 1;
+    if (biny < 1) biny = 1;
+    if (binx > nx) binx = nx;
+    if (biny > ny) biny = ny;
+
+    sferr = h_eff_SF_err_->GetBinContent (binx, biny);
+  }
+  else
+  {
+    if (fabs(eta) > 2.5)
+    {
+      int binglobal = h_mistag_SF_err_->FindBin (pt, eta);
+      int binx, biny, binz;
+      h_mistag_SF_err_->GetBinXYZ (binglobal, binx, biny, binz); // converts to x, y bins
+      int nx = h_mistag_SF_err_->GetNbinsX();
+      int ny = h_mistag_SF_err_->GetNbinsY();
+
+      // under-overflows
+      if (binx < 1) binx = 1;
+      if (biny < 1) biny = 1;
+      if (binx > nx) binx = nx;
+      if (biny > ny) biny = ny;
+
+      sferr = h_mistag_SF_->GetBinContent (binx, biny);
+    }
+    else /* fabs(eta) < 2.5 */
+    {
+      // inside tracker acceptance use the SF itself as error on the mistag
+      sferr = getSF(isReal, pt, eta);
+    }
+  }
+
+  return sferr;
+}
+
+//getEvtWeight - Returns a vector with {central SF, SF_up, SF_down}
+std::vector<float> PuJetIdSF::getEvtWeight(bigTree &theBigTree, TLorentzVector tau1, TLorentzVector tau2)
 {
   // Weight for each event
-  float eventWeight;
+  std::vector<float> eventWeight(3);
   float P_MC   = 1.;
   float P_DATA = 1.;
```

Comment:
Why there is no "P_MC_up" and "P_MC_down"?

Reply:
Becayse only the "sf" is shifted, not the "eff", and P_MC only depend on the "eff".